eZ/Bundle/EzPublishCoreBundle/SiteAccess/Config/ComplexConfigProcessor.php

```diff
@@ -39,31 +46,35 @@ public function processComplexSetting(string $setting): string
             throw new ParameterNotFoundException($setting, null, [$siteAccessName]);
         }
 
-        $complexSettingParser = new ComplexSettingParser();
         $settingValue = $this->configResolver->getParameter($setting, null, $siteAccessName);
 
-        if (!$complexSettingParser->containsDynamicSettings($settingValue)) {
+        if (!$this->complexSettingParser->containsDynamicSettings($settingValue)) {
             return $settingValue;
         }
 
         // we kind of need to process this as well, don't we ?
-        if ($complexSettingParser->isDynamicSetting($settingValue)) {
-            $parts = $complexSettingParser->parseDynamicSetting($settingValue);
+        if ($this->complexSettingParser->isDynamicSetting($settingValue)) {
+            $parts = $this->complexSettingParser->parseDynamicSetting($settingValue);
 
             return $this->configResolver->getParameter($parts['param'], null, $siteAccessName);
         }
 
-        $value = $settingValue;
-        foreach ($complexSettingParser->parseComplexSetting($settingValue) as $dynamicSetting) {
-            $parts = $complexSettingParser->parseDynamicSetting($dynamicSetting);
+        return $this->processSettingValue($settingValue);
+    }
+
+    public function processSettingValue(string $value): string
+    {
+        foreach ($this->complexSettingParser->parseComplexSetting($value) as $dynamicSetting) {
+            $parts = $this->complexSettingParser->parseDynamicSetting($dynamicSetting);
             if (!isset($parts['namespace'])) {
                 $parts['namespace'] = self::DEFAULT_NAMESPACE;
             }
-            if (!isset($parts['scope'])) {
-                $parts['scope'] = $siteAccessName;
-            }
 
-            $dynamicSettingValue = $this->configResolver->getParameter($parts['param'], $parts['namespace'], $parts['scope']);
+            $dynamicSettingValue = $this->configResolver->getParameter(
+                $parts['param'],
+                $parts['namespace'],
+                $parts['scope'] ?? $this->siteAccessService->getCurrent()->name
```

Comment:
Kept for BC of behavior, but this is indeed redundant (TBD if can be safely fixed).